From f9b77e037d6f86b33557ec1bd2dd3ae3f1c25025 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Tue, 26 Nov 2024 14:00:58 +0100
Subject: [PATCH 708/841] feat(st-clk): add ck_sys_dbg clock for STM32MP2

Add management of ck_sys_dbg clock.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git f0e62c74a123eae8e906862ba3e03f34e11288f5 feat(st-clk): add ck_sys_dbg clock for STM32MP2]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: Ic74d31a82375c6361689f49f41a7240714c6ac10
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/431875
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
---
 drivers/st/clk/clk-stm32mp2.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/st/clk/clk-stm32mp2.c b/drivers/st/clk/clk-stm32mp2.c
index fcdead227..e93e959ea 100644
--- a/drivers/st/clk/clk-stm32mp2.c
+++ b/drivers/st/clk/clk-stm32mp2.c
@@ -274,6 +274,7 @@ enum clock {
 #endif /* !STM32MP21 */
 	_CK_STGEN,
 	_CK_SYSCPU1,
+	_CK_SYSDBG,
 	_CK_SYSRAM,
 	_CK_UART4,
 	_CK_UART5,
@@ -444,6 +445,7 @@ enum enum_gate_cfg {
 	GATE_IWDG1,
 	GATE_IWDG2,
 
+	GATE_DBG,
 	GATE_DDRCAPB,
 	GATE_DDR,
 
@@ -580,6 +582,7 @@ static const struct gate_cfg gates_mp2[LAST_GATE] = {
 	GATE_CFG(GATE_BSEC,		RCC_BSECCFGR,		1,	0),
 	GATE_CFG(GATE_IWDG1,		RCC_IWDG1CFGR,		1,	0),
 	GATE_CFG(GATE_IWDG2,		RCC_IWDG2CFGR,		1,	0),
+	GATE_CFG(GATE_DBG,		RCC_DBGCFGR,		8,	0),
 	GATE_CFG(GATE_DDRCAPB,		RCC_DDRCAPBCFGR,	1,	0),
 	GATE_CFG(GATE_DDR,		RCC_DDRCFGR,		1,	0),
 	GATE_CFG(GATE_USART2,		RCC_USART2CFGR,		1,	0),
@@ -1643,6 +1646,8 @@ static const struct clk_stm32 stm32mp2_clk[CK_LAST] = {
 	STM32_GATE(_CK_DDRCAPB, CK_BUS_DDRC, _CK_ICN_APB4, 0, GATE_DDRCAPB),
 	STM32_GATE(_CK_DDR, CK_BUS_DDRCFG, _CK_ICN_APB4, 0, GATE_DDR),
 
+	STM32_GATE(_CK_SYSDBG, CK_SYSDBG, _CK_ICN_APBDBG, 0, GATE_DBG),
+
 	STM32_GATE(_CK_USART2, CK_KER_USART2, _CK_FLEXGEN_08, 0, GATE_USART2),
 	STM32_GATE(_CK_UART4, CK_KER_UART4, _CK_FLEXGEN_08, 0, GATE_UART4),
 	STM32_GATE(_CK_USART3, CK_KER_USART3, _CK_FLEXGEN_09, 0, GATE_USART3),
-- 
2.47.2

